bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitOHItemProxy.java

```diff
@@ -0,0 +1,156 @@
+            // - DIMMER_MODE_FILTER_ON_EXCEPT100 is not enabled and brightness is null or below 100
+            if ((on == OnOffType.OFF) || (dimmerMode == DIMMER_MODE_NONE)
+                    || (dimmerMode == DIMMER_MODE_FILTER_BRIGHTNESS_100)
+                    || ((dimmerMode == DIMMER_MODE_FILTER_ON_EXCEPT_BRIGHTNESS_100)
+                            && ((brightness == null) || (brightness.intValue() == 100)))) {
+                logger.trace("send OnOff command for item {} with value {}", item, on);
+                ((DimmerItem) item).send(on);
+            }
+        }
+
+        // if hue or saturation present, send an HSBType state update. no filter applied for HUE & Saturation
+        if ((hue != null) || (saturation != null)) {
+            if (item instanceof ColorItem) {
+                // logic for ColorItem = combine hue, saturation and brightness update to one command
+                final HSBType currentState = item.getState() instanceof UnDefType ? HSBType.BLACK
+                        : (HSBType) item.getState();
+                ((ColorItem) item).send(new HSBType(hue != null ? hue : currentState.getHue(),
+                        saturation != null ? saturation : currentState.getSaturation(),
+                        brightness != null ? brightness : currentState.getBrightness()));
+                logger.trace("send HSB command for item {} with following values hue={} saturation={} brightness={}",
+                        item, hue, saturation, brightness);
+            }
+        } else if ((brightness != null) && (item instanceof DimmerItem)) {
+            // sends brightness:
+            // - DIMMER_MODE_NONE
+            // - DIMMER_MODE_FILTER_ON
+            // - other modes (DIMMER_MODE_FILTER_BRIGHTNESS_100 or DIMMER_MODE_FILTER_ON_EXCEPT_BRIGHTNESS_100) and
+            // <100%.
+            if ((dimmerMode == DIMMER_MODE_NONE) || (dimmerMode == DIMMER_MODE_FILTER_ON)
+                    || (brightness.intValue() < 100)) {
+                logger.trace("send brightness command for item {} with value {}", item, brightness);
+                ((DimmerItem) item).send(brightness);
+            }
+        }
+        commandCache.clear();
+    }
+
+    public synchronized void sendCommandProxy(final String commandType, final State state) {
+        commandCache.put(commandType, state);
+        logger.trace("add command to command cache: item {}, command type {}, command state {}. cache state after: {}",
+                this, commandType, state, commandCache);
+
+        // if cache has already HUE+SATURATION or BRIGHTNESS+ON then we don't expect any further relevant command
+        if ((commandCache.containsKey(HUE_COMMAND) && commandCache.containsKey(SATURATION_COMMAND))
+                || (commandCache.containsKey(BRIGHTNESS_COMMAND) && commandCache.containsKey(ON_COMMAND))) {
+            if (future != null)
+                future.cancel(true);
```

Comment:
I'm not sure how much good cancelling with interrupt would do here, might cause more problems than if you just let the future complete if it was already running.
```suggestion
                future.cancel(false);
```

Reply:
it should make much difference in that case. we would interrupt "sendCommand" which we will re-run directly after cancelation. but you are right, false is probably the saver option.